plugins/k8saudit/pkg/k8saudit/source.go

```diff
@@ -50,11 +56,47 @@ func (k *Plugin) Open(params string) (source.Instance, error) {
 	case "https":
 		return k.OpenWebServer(u.Host, u.Path, true)
 	case "": // by default, fallback to opening a filepath
-		file, err := os.Open(params)
+		trimmed := strings.TrimSpace(params)
+
+		if strings.HasSuffix(trimmed, ".log") {
+			file, err := os.Open(trimmed)
+			if err != nil {
+				return nil, err
+			}
+			return k.OpenReader(file)
+		}
+
+		files, err := ioutil.ReadDir(trimmed)
 		if err != nil {
 			return nil, err
 		}
-		return k.OpenReader(file)
+
+		// pointer to the struct to be able to modify it afterwards
+		k8saudit := make(map[int]*AuditMap)
+		numFiles := 0
+
+		for _, f := range files {
+			fmt.Println(f.Name())
+			if !f.IsDir() && strings.HasSuffix(f.Name(), ".log") {
+				var auditFile io.Reader
+				auditFile, err := os.Open(f.Name())
+				if err != nil {
+					return nil, err
+				}
+				logFile := &AuditMap{fileName: f.Name(), modified: f.ModTime(), pointer: auditFile}
+				k8saudit[numFiles] = logFile
+				numFiles += 1
+			}
+		}
+
+		results := []io.Reader
+		for i := 0; i < len(k8saudit); i++ {
+			results = append(results, k8saudit[i].pointer)
+		}
+		// results... = type []any
+		AllAuditFiles := io.MultiReader(results...)
+		return k.OpenReader(AllAuditFiles)
```

Comment:
This block can be simplified as follows:
```suggestion
		// sort files by modificationt time
		sort.Slice(files, func(i, j int) bool {
			return files[i].ModTime().Before(files[j].ModTime())
		})

		// open all files as reader
		results := []io.Reader
		for _, f := range files {
			if !f.IsDir() {
				auditFile, err := os.Open(f.Name())
				if err != nil {
					return nil, err
				}
				results = append(results, auditFile)
			}
		}

		// concat all readers into a multi-reader
		return k.OpenReader(io.MultiReader(results...))
```

Also, be mindful that `k.OpenReader` will expect an `io.ReadCloser`, which the result of `io.MultiReader` does not implement. Did you test that this works?

Reply:
Hey, Jason, I thought that the file names would have the ".log" extension by default, but I understand that you may not want to limit it in such a way. I used `io.NopCloser()` to wrap the readers so it should satisfy `k.OpenReader`. 
So far, I have only tested this logic on dummy JSONL files - tomorrow I can test it in an actual cluster.

Reply:
One thing that worries me is that when files are merged, the MultiReader concatenates the last JSON object with the first JSON object from the next file which results in two objects on the same line. This breaks the JSONL convention and may cause Falco to error during parsing. 
`{"year":"2018","host":"Russia","winner":"France"}
{"year":"2014","host":"Brazil","winner":"Germany"}
{"year":"2010","host":"South Africa","winner":"Spain"}
{"year":"2006","host":"Germany","winner":"Italy"}{"year":"2018","host":"Russia","winner":"France"}
{"year":"2014","host":"Brazil","winner":"Germany"}
{"year":"2010","host":"South Africa","winner":"Spain"}
{"year":"2006","host":"Germany","winner":"Italy"}`